src/dev/register_git_hook/register_git_hook.js

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import chalk from 'chalk';
+import { chmod, lstat, stat, symlink, unlink, writeFile } from 'fs';
+import { trim } from 'lodash';
+import { resolve } from 'path';
+import { promisify } from 'util';
+import { REPO_ROOT } from '../constants';
+
+const chmodAsync = promisify(chmod);
+const lstatAsync = promisify(lstat);
+const statAsync = promisify(stat);
+const symlinkAsync = promisify(symlink);
+const unlinkAsync = promisify(unlink);
+const writeFileAsync = promisify(writeFile);
+
+const KBN_PRECOMMIT_GIT_HOOK_SCRIPT_PATH = resolve(REPO_ROOT, '.git/hooks/kbn_precommit_hook');
+const PRECOMMIT_GIT_HOOK_SCRIPT_PATH = resolve(REPO_ROOT, '.git/hooks/pre-commit');
+
+function getKbnPrecommitGitHookScript(rootPath, kbnPrecommitGitHookScriptPath) {
+  return trim(`
```

Comment:
I don't think trim is going to work here - might need to use something like `dedent` to remove the leading spaces on the lines.  We have used this in the past in the kbn-pm and kbn-es packages.

Reply:
I'll take a look 😃